### PR TITLE
[doc] Fix broken URLs

### PR DESCRIPTION
**Pull request by vvv**
Use proper syntax when [linking to items by name].

Before:
https://docs.rs/arrow-array/latest/arrow_array/iterator/%5Bcrate::PrimitiveArray%5D

After:
https://docs.rs/arrow-array/latest/arrow_array/array/struct.PrimitiveArray.html

[linking to items by name]: https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html

### CODE REVIEW

**Review by tustvold — APPROVED** [review]
Oops, thank you 👍

File: arrow-array/src/types.rs [hunk 1]
```diff
@@ -565,9 +565,9 @@ mod decimal {
 /// Implemented by [`Decimal128Type`] and [`Decimal256Type`] for [`Decimal128Array`]
 /// and [`Decimal256Array`] respectively
 ///
-/// [`DecimalArray`]: [crate::array::DecimalArray]
-/// [`Decimal128Array`]: [crate::array::Decimal128Array]
-/// [`Decimal256Array`]: [crate::array::Decimal256Array]
+/// [`DecimalArray`]: crate::array::DecimalArray
```
Comment (vvv) [inline arrow-array/src/types.rs:568]: [CI is unhappy](https://github.com/apache/arrow-rs/actions/runs/3862014965/jobs/6583385378#step:6:303), for there is no such identifier.
```
error: unresolved link to `crate::array::DecimalArray`
   --> arrow-array/src/types.rs:568:23
    |
568 | /// [`DecimalArray`]: crate::array::DecimalArray
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `DecimalArray` in module `array`

error: could not document `arrow-array`
```

Should it be `PrimitiveArray`?

I found #2637, it seems to be related...

File: arrow-array/src/iterator.rs [hunk 2]
```diff
@@ -39,9 +39,9 @@ use crate::array::{
 /// there are more efficient ways to iterate over just the non-null indices, this functionality
 /// is provided by [`compute::try_unary`]
 ///
-/// [`PrimitiveArray`]: [crate::PrimitiveArray]
-/// [`compute::unary`]: [arrow::compute::unary]
-/// [`compute::try_unary`]: [arrow::compute::try_unary]
+/// [`PrimitiveArray`]: crate::PrimitiveArray
+/// [`compute::unary`]: https://docs.rs/arrow/latest/arrow/compute/fn.unary.html
+/// [`compute::try_unary`]: https://docs.rs/arrow/latest/arrow/compute/fn.try_unary.html
```
Comment (vvv) [inline arrow-array/src/iterator.rs:44]: `arrow` crate is not among `arrow-array`'s dependencies, so its items cannot be referred to using [‘intra-doc link’] syntax.

[‘intra-doc link’]: https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html

File: arrow-array/src/types.rs [hunk 3]
```diff
@@ -559,15 +559,15 @@ mod decimal {
     impl DecimalTypeSealed for Decimal256Type {}
 }
 
-/// A trait over the decimal types, used by [`DecimalArray`] to provide a generic
+/// A trait over the decimal types, used by [`PrimitiveArray`] to provide a generic
```
Comment (vvv) [inline arrow-array/src/types.rs:562]: @tustvold Is this change correct?

**Review by tustvold — APPROVED** [review]

**Pull request merged**